From ea7f537db6cc0edd8d497e66255bec448c922639 Mon Sep 17 00:00:00 2001
From: "David C. Manuelda" <StormByte@gmail.com>
Date: Sun, 9 Aug 2026 06:05:20 +0200
Subject: [PATCH] PR #1476

Signed-off-by: David C. Manuelda <StormByte@gmail.com>
---
 libvmaf/src/feature/integer_vif.c | 34 +++++++++++++++----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/libvmaf/src/feature/integer_vif.c b/libvmaf/src/feature/integer_vif.c
index b3a6a89..385e9cc 100644
--- a/libvmaf/src/feature/integer_vif.c
+++ b/libvmaf/src/feature/integer_vif.c
@@ -641,22 +641,22 @@ static int init(VmafFeatureExtractor *fex, enum VmafPixelFormat pix_fmt,
     if (!data) return -ENOMEM;
     memset(data, 0, data_sz);
 
-    s->public.buf.data = data; data += pad_size;
-    s->public.buf.ref = data; data += frame_size + pad_size + pad_size;
-    s->public.buf.dis = data; data += frame_size + pad_size;
-    s->public.buf.mu1 = data; data += h * s->public.buf.stride_16;
-    s->public.buf.mu2 = data; data += h * s->public.buf.stride_16;
-    s->public.buf.mu1_32 = data; data += s->public.buf.stride_32;
-    s->public.buf.mu2_32 = data; data += s->public.buf.stride_32;
-    s->public.buf.ref_sq = data; data += s->public.buf.stride_32;
-    s->public.buf.dis_sq = data; data += s->public.buf.stride_32;
-    s->public.buf.ref_dis = data; data += s->public.buf.stride_32;
-    s->public.buf.tmp.mu1 = data; data += s->public.buf.stride_tmp;
-    s->public.buf.tmp.mu2 = data; data += s->public.buf.stride_tmp;
-    s->public.buf.tmp.ref = data; data += s->public.buf.stride_tmp;
-    s->public.buf.tmp.dis = data; data += s->public.buf.stride_tmp;
-    s->public.buf.tmp.ref_dis = data; data += s->public.buf.stride_tmp;
-    s->public.buf.tmp.ref_convol = data; data += s->public.buf.stride_tmp;
+    s->public.buf.data = data; data = (char *)data + pad_size;
+    s->public.buf.ref = data; data = (char *)data + (frame_size + pad_size + pad_size);
+    s->public.buf.dis = data; data = (char *)data + (frame_size + pad_size);
+    s->public.buf.mu1 = data; data = (char *)data + (h * s->public.buf.stride_16);
+    s->public.buf.mu2 = data; data = (char *)data + (h * s->public.buf.stride_16);
+    s->public.buf.mu1_32 = data; data = (char *)data + s->public.buf.stride_32;
+    s->public.buf.mu2_32 = data; data = (char *)data + s->public.buf.stride_32;
+    s->public.buf.ref_sq = data; data = (char *)data + s->public.buf.stride_32;
+    s->public.buf.dis_sq = data; data = (char *)data + s->public.buf.stride_32;
+    s->public.buf.ref_dis = data; data = (char *)data + s->public.buf.stride_32;
+    s->public.buf.tmp.mu1 = data; data = (char *)data + s->public.buf.stride_tmp;
+    s->public.buf.tmp.mu2 = data; data = (char *)data + s->public.buf.stride_tmp;
+    s->public.buf.tmp.ref = data; data = (char *)data + s->public.buf.stride_tmp;
+    s->public.buf.tmp.dis = data; data = (char *)data + s->public.buf.stride_tmp;
+    s->public.buf.tmp.ref_dis = data; data = (char *)data + s->public.buf.stride_tmp;
+    s->public.buf.tmp.ref_convol = data; data = (char *)data + s->public.buf.stride_tmp;
     s->public.buf.tmp.dis_convol = data;
 
     s->feature_name_dict =
@@ -667,7 +667,7 @@ static int init(VmafFeatureExtractor *fex, enum VmafPixelFormat pix_fmt,
     return 0;
 
 fail:
-    if (data) aligned_free(data);
+    if (s->public.buf.data) aligned_free(s->public.buf.data);
     vmaf_dictionary_free(&s->feature_name_dict);
     return -ENOMEM;
 }
-- 
2.50.1 (Apple Git-155)

